### PR TITLE
[ALT-10963] Change `check_profile` , opening project view.

### PR DESCRIPTION
**Pull request by KirillovProj**
## Summary
- preserve `check_profile` for remote tasks during YAML serialization/deserialization and student task updates, so server data is not overwritten with an empty value
- harden Hyperskill project opening and YAML loading by tolerating missing config directories, skipping unsupported topic tasks, and handling remote YAML loading failures per item instead of aborting the whole load
- fix task directory resolution in the course view so projects still open correctly when `Task.getDir()` cannot resolve the task directory
- block remote task submission when `check_profile` is empty and add logging around load/apply/save/submit flows to make these failures diagnosable

## Details
- added a dedicated YAML mixin for `RemoteEduTask` and explicitly serialize `check_profile`
- excluded `check_profile` from generic additional-property serialization to avoid duplicate/conflicting YAML output
- changed student task change application to update `check_profile` only for remote tasks and log mismatches
- made config-dir lookups nullable and updated YAML sync/load code to fail soft when files or directories are missing
- updated deep remote info loading to continue after `RemoteYamlLoadingException`
- adjusted course view logic to resolve visible task files from the actual task directory instead of relying on a later `getDir()` lookup
- added regression coverage for YAML serialization, YAML change handling, and course view task directory lookup
- updated the Hyperskill courses link in the new project UI to use the tracked campaign URL

## Testing
- added/updated regression tests for:
  - remote task YAML serialization
  - YAML change-after-event handling
  - course view task directory lookup

### CODE REVIEW

**Review by Tsyklop — APPROVED** [review]

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt [hunk 1]
```diff
@@ -24,11 +28,14 @@ class StudentTaskChangeApplier(project: Project) : TaskChangeApplier(project) {
...
-    when (existingItem) {
-      is EduTask -> {
-        if (existingItem is RemoteEduTask) {
-          existingItem.checkProfile = (deserializedItem as RemoteEduTask).checkProfile
-        }
+    if (existingItem is RemoteEduTask && deserializedItem is RemoteEduTask) {
+      val newCheckProfile = deserializedItem.checkProfile
+      if (newCheckProfile != existingItem.checkProfile) {
+        LOG.info("Updating checkProfile for task ${existingItem.name}: '${existingItem.checkProfile}' -> '$newCheckProfile'")
+        existingItem.checkProfile = newCheckProfile
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt:28]: **Blocking — the original bug is back.**

The original fix in commit `3766a0d` was `if (newCheckProfile.isNotEmpty()) existingItem.checkProfile = newCheckProfile`. Commit `e44e4073` replaced that with a plain inequality check, so now if the deserialized YAML carries `check_profile: ''` while the in-memory task has a real profile (e.g. `"hyperskill_go"`), this line still overwrites the existing value with an empty string — the exact scenario ALT-10963 is supposed to prevent. The `LOG.warn("checkProfile is empty … after applying changes")` below is essentially logging the bug rather than preventing it.

Suggested fix: bring back the guard, e.g.
```kotlin
if (newCheckProfile.isNotEmpty() && newCheckProfile != existingItem.checkProfile) {
  existingItem.checkProfile = newCheckProfile
}
```
and add a regression test that loads YAML with an empty `check_profile` over a `RemoteEduTask` whose `checkProfile` was already set.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt [hunk 2]
```diff
@@ -11,9 +12,12 @@ import org.hyperskill.academy.learning.messages.EduCoreBundle
 import org.hyperskill.academy.learning.yaml.errorHandling.YamlLoadingException
 import org.hyperskill.academy.learning.yaml.format.TaskChangeApplier
 
-
 class StudentTaskChangeApplier(project: Project) : TaskChangeApplier(project) {
+  private val LOG = logger<StudentTaskChangeApplier>()
   override fun applyChanges(existingItem: Task, deserializedItem: Task) {
+    LOG.info("Applying changes for task: ${existingItem.name} (id=${existingItem.id})")
+    LOG.info("Existing item checkProfile: ${(existingItem as? RemoteEduTask)?.checkProfile}")
+    LOG.info("Deserialized item checkProfile: ${(deserializedItem as? RemoteEduTask)?.checkProfile}")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt]: Three `LOG.info` lines per `applyChanges` call will be very noisy on real courses (this runs for every YAML change event). Please downgrade to `debug` or drop them — the same information is already logged on the setter and on save.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt [hunk 3]
```diff
@@ -24,11 +28,14 @@ class StudentTaskChangeApplier(project: Project) : TaskChangeApplier(project) {
...
-          existingItem.checkProfile = (deserializedItem as RemoteEduTask).checkProfile
-        }
+    if (existingItem is RemoteEduTask && deserializedItem is RemoteEduTask) {
+      val newCheckProfile = deserializedItem.checkProfile
+      if (newCheckProfile != existingItem.checkProfile) {
+        LOG.info("Updating checkProfile for task ${existingItem.name}: '${existingItem.checkProfile}' -> '$newCheckProfile'")
+        existingItem.checkProfile = newCheckProfile
+      }
+      if (existingItem.checkProfile.isEmpty()) {
+        LOG.warn("checkProfile is empty for RemoteEduTask ${existingItem.name} after applying changes")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/format/student/StudentTaskChangeApplier.kt]: Once the empty-overwrite guard from line 35 is restored, this warning becomes unreachable / misleading. Either remove it, or replace it with a hard failure (`throw YamlLoadingException(...)`) so we don't silently keep a task in a broken state.

File: hs-edu-format/src/org/hyperskill/academy/learning/courseFormat/tasks/RemoteEduTask.kt [hunk 4]
```diff
@@ -17,6 +22,12 @@ class RemoteEduTask : EduTask {
    * MUST NOT BE HARDCODED
    */
   var checkProfile: String = ""
+    set(value) {
+      if (field != value) {
+        LOG.info("[DEBUG_LOG] RemoteEduTask.checkProfile set to '$value' (was '$field')")
```
Comment (meanmail) [inline hs-edu-format/src/org/hyperskill/academy/learning/courseFormat/tasks/RemoteEduTask.kt]: Putting a logging side-effect on a domain-model property setter (and with a `[DEBUG_LOG]` prefix) reads like leftover debugging instrumentation. If we need to trace when `checkProfile` changes, do it at the call sites (loader / change applier) rather than the model, and at `debug` level.

File: hs-edu-format/src/org/hyperskill/academy/learning/courseFormat/tasks/RemoteEduTask.kt [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+import org.hyperskill.academy.learning.courseFormat.logger
 import org.jetbrains.annotations.NonNls
 import java.util.*
 
 /**
  * By default, EduTask is checked with local tests, but if `is_remote_tested` flag is true, then we should send code to remote check
  */
 // see EDU-4504 Support Go edu problems
+
+
```
Comment (meanmail) [inline hs-edu-format/src/org/hyperskill/academy/learning/courseFormat/tasks/RemoteEduTask.kt]: Nit: two consecutive blank lines between the type-level comment and `private val LOG`.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt [hunk 6]
```diff
@@ -2,16 +2,21 @@ package org.hyperskill.academy.learning.yaml
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.google.common.annotations.VisibleForTesting
+import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.fileEditor.FileEditorManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.util.messages.Topic
 import org.hyperskill.academy.learning.*
+import org.hyperskill.academy.learning.api.EduOAuthCodeFlowConnector
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt]: `EduOAuthCodeFlowConnector` is imported but not referenced anywhere in this file.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt [hunk 7]
```diff
@@ -24,14 +29,21 @@ import org.hyperskill.academy.learning.yaml.errorHandling.*
 import org.hyperskill.academy.learning.yaml.format.YamlMixinNames.TASK
 import org.hyperskill.academy.learning.yaml.format.getChangeApplierForItem
 import org.jetbrains.annotations.NonNls
+import java.net.HttpURLConnection.HTTP_BAD_REQUEST
+import java.net.HttpURLConnection.HTTP_FORBIDDEN
+import java.net.HttpURLConnection.HTTP_UNAUTHORIZED
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt]: `HTTP_BAD_REQUEST`, `HTTP_FORBIDDEN`, `HTTP_UNAUTHORIZED` are imported but never used in this file — please remove.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt [hunk 8]
```diff
@@ -55,6 +67,9 @@ object YamlLoader {
 
     val existingItem = getStudyItemForConfig(project, configFile)
     val deserializedItem = deserializeItemProcessingErrors(configFile, project, loadFromVFile, mapper) ?: return
+    if (deserializedItem is Task) {
+      com.intellij.openapi.diagnostic.logger<YamlLoader>().info("Deserialized task ${deserializedItem.name} (type=${deserializedItem.itemType}), checkProfile='${(deserializedItem as? RemoteEduTask)?.checkProfile}'")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt]: `LOG` is already declared at line 42 of the same object — use it instead of creating an ad-hoc `com.intellij.openapi.diagnostic.logger<YamlLoader>()` here. Also, logging every deserialized task at `info` is noisy; consider `debug`.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt [hunk 9]
```diff
@@ -217,6 +238,12 @@ object YamlLoader {
 
 }
 
+@PublishedApi
+internal fun Lesson.isHyperskillTopicsLesson(): Boolean {
+  val section = parentOrNull as? Section ?: return false
+  return course is HyperskillCourse && section.presentableName == EduFormatNames.HYPERSKILL_TOPICS
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlLoader.kt]: Matching by `section.presentableName == "Topics"` is fragile (it is a display string and could in principle be localized or renamed). `HyperskillCourse` already exposes `getTopicsSection()` — prefer reusing it and comparing the section reference:
```kotlin
val course = course as? HyperskillCourse ?: return false
return (parentOrNull as? Section) === course.getTopicsSection()
```

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlFormatSynchronizer.kt [hunk 10]
```diff
@@ -185,6 +180,7 @@ object YamlFormatSynchronizer {
             )
           }
           val yamlText = mapper.writeValueAsString(this)
+          LOG.info("Saving item ${this.name} to ${file.name}. YAML content snippet: ${yamlText.take(200)}")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/yaml/YamlFormatSynchronizer.kt]: Logging the first 200 chars of every YAML write at `info` will produce a large volume of log lines on real courses (every task save, every status change). Please drop or downgrade to `debug`.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/checker/HyperskillSubmitConnector.kt [hunk 11]
```diff
@@ -79,6 +79,12 @@ object HyperskillSubmitConnector {
   }
 
   fun submitRemoteEduTask(task: RemoteEduTask, files: List<SolutionFile>): Result<StepikBasedSubmission, String> {
+    LOG.info("Submitting RemoteEduTask: ${task.name} (id=${task.id}), checkProfile='${task.checkProfile}'")
+    if (task.checkProfile.isEmpty()) {
+      val message = "Check profile is empty for task ${task.name}. Submission aborted to avoid server error."
+      LOG.error(message)
+      return Err(message + " Please try to 'Synchronize project' or contact support.")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/checker/HyperskillSubmitConnector.kt]: User-facing string is hardcoded in English here. Other error strings in this file go through `EduCoreBundle.message(...)` — please follow the same convention so the message stays translatable.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/checker/HyperskillSubmitConnector.kt [hunk 12]
```diff
@@ -79,6 +79,12 @@ object HyperskillSubmitConnector {
   }
 
   fun submitRemoteEduTask(task: RemoteEduTask, files: List<SolutionFile>): Result<StepikBasedSubmission, String> {
+    LOG.info("Submitting RemoteEduTask: ${task.name} (id=${task.id}), checkProfile='${task.checkProfile}'")
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/checker/HyperskillSubmitConnector.kt]: Logging every submission at `info` is fine for now, but please consider whether `debug` would be enough — every check action will land here.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/submissions/HyperskillSubmissionFactory.kt [hunk 13]
```diff
@@ -20,16 +20,18 @@ object HyperskillSubmissionFactory {
 
   fun createEduTaskSubmission(task: Task, attempt: Attempt, files: List<SolutionFile>, feedback: String): StepikBasedSubmission {
     val reply = EduTaskReply()
-    reply.feedback = Feedback(feedback)
     reply.score = if (task.status == CheckStatus.Solved) "1" else "0"
     reply.solution = files
+    reply.feedback = Feedback(feedback)
+    reply.checkProfile = ""
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/submissions/HyperskillSubmissionFactory.kt]: `createEduTaskSubmission` is for plain `EduTask`, which has no notion of `checkProfile`. Forcing `reply.checkProfile = ""` here sends an empty `check_profile` to the server for every regular edu submission — please confirm the server tolerates it; otherwise drop the assignment. If it is needed only to satisfy some serialization contract, a comment explaining why would help.

File: intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/api/HyperskillConnector.kt [hunk 14]
```diff
@@ -81,6 +82,14 @@ abstract class HyperskillConnector : EduOAuthCodeFlowConnector<HyperskillAccount
     val request = chain.request()
     val newUrl = request.url.newBuilder().addQueryParameter("ide_rpc_port", BuiltInServerManager.getInstance().port.toString()).build()
     val newRequest = request.newBuilder().url(newUrl).build()
+
+    if (newRequest.body != null) {
+      val buffer = Buffer()
+      runCatching { newRequest.body!!.writeTo(buffer) }
+        .onSuccess { LOG.info("Retrofit request body for ${newRequest.method} ${newRequest.url}: ${buffer.readUtf8()}") }
```
Comment (meanmail) [inline intellij-plugin/hs-core/src/org/hyperskill/academy/learning/stepik/hyperskill/api/HyperskillConnector.kt]: Logging the full request body at `info` for **every** Retrofit call is risky: it will (a) flood `idea.log` and (b) potentially log payloads that include tokens/PII (this interceptor runs after the auth interceptors). At minimum downgrade to `debug` and gate by a system property; ideally remove before merge.

File: hs-edu-format/src/org/hyperskill/academy/learning/yaml/format/remote/RemoteEduTaskYamlMixin.kt [hunk 15]
```diff
@@ -16,6 +17,8 @@ import org.hyperskill.academy.learning.yaml.format.student.StudentTaskYamlMixin
 @Suppress("unused")
 @JsonPropertyOrder(TYPE, CUSTOM_NAME, FILES, FEEDBACK_LINK, CHECK_PROFILE, STATUS, FEEDBACK, RECORD, TAGS)
 class RemoteEduTaskYamlMixin : StudentTaskYamlMixin() {
-  @JsonProperty(CHECK_PROFILE)
-  private lateinit var checkProfile: String
+  @get:JsonProperty(CHECK_PROFILE)
+  @set:JsonProperty(CHECK_PROFILE)
+  @get:JsonInclude(JsonInclude.Include.ALWAYS)
+  var checkProfile: String = ""
 }
```
Comment (meanmail) [inline hs-edu-format/src/org/hyperskill/academy/learning/yaml/format/remote/RemoteEduTaskYamlMixin.kt]: Nit: file is missing a final newline.

**Review by meanmail — CHANGES_REQUESTED** [review]
## Review summary

The PR has grown well beyond its original one-line scope. The original empty-string-overwrite fix from commit `3766a0d` was lost in subsequent commits, so **the very bug this PR was created for is no longer prevented**. There are also several leftover debug logs, unused imports, scope creep (UI URL change, project view refactor, soft-fail YAML loading, Retrofit body logging, two large docs files), and a few smaller correctness/style issues. Details inline.

### Blocking
1. `StudentTaskChangeApplier` still overwrites a non-empty `checkProfile` with an empty value when the deserialized item carries `""`. The `newCheckProfile.isNotEmpty()` guard from `3766a0d` was dropped in `e44e4073`. Please restore it and add a regression test covering exactly this scenario.

### Strongly recommended before merge
2. Strip the `[DEBUG_LOG]`-prefixed and per-call info logs (model setter, every YAML save, every `applyChanges`, every Retrofit request body, every deserialization). At minimum downgrade them to `debug` — `LOG.info` here will produce a lot of noise on real courses and the Retrofit body log can leak auth-bearing payloads.
3. Drop unused imports in `YamlLoader.kt` (`EduOAuthCodeFlowConnector`, `HTTP_BAD_REQUEST`, `HTTP_FORBIDDEN`, `HTTP_UNAUTHORIZED`).
4. Consider splitting the unrelated changes (course-view refactor, UTM-link change, soft-fail in `YamlDeepLoader`, Retrofit body logging, the two new docs files) into separate PRs — they are not part of ALT-10963 and make review harder.

### Smaller things
5. `isHyperskillTopicsLesson()` matches `section.presentableName == "Topics"` — fragile; reuse `HyperskillCourse.getTopicsSection()` and compare by reference.
6. `HyperskillSubmitConnector` user-facing error string is hardcoded English instead of `EduCoreBundle`.
7. `HyperskillSubmissionFactory.createEduTaskSubmission` now explicitly assigns `reply.checkProfile = ""`. `EduTask` has no check profile concept — please confirm the server expects this field on plain edu submissions; otherwise drop the assignment.
8. `RemoteEduTaskYamlMixin.kt` is missing a final newline; two blank lines before `private val LOG` in `RemoteEduTask.kt`.

**Review by meanmail — APPROVED** [review]
Thanks for the quick turnaround! All blocking and important comments from my previous review are addressed in `0ecdc84` — the `isNotEmpty()` guard is back in `StudentTaskChangeApplier`, two targeted regression tests cover the empty/non-empty cases, all the `info`-level diagnostic logs are gone (Retrofit body, model setter, per-deserialization, per-save, per-applyChanges), unused imports in `YamlLoader` are dropped, `isHyperskillTopicsLesson` now uses `HyperskillCourse.getTopicsSection()` by reference, the hardcoded English error string is replaced with a new `EduCoreBundle` key, and the spurious `reply.checkProfile = ""` in `createEduTaskSubmission` is removed. Final newlines also fixed. LGTM. (Couldn't resolve the inline threads myself — missing permission — please mark them as resolved in the UI.)
